x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/sender/RequestExecutorServiceSettings.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.http.sender;
+
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class RequestExecutorServiceSettings {
+
+    /**
+     * The capacity of the internal queue. Zero is considered unlimited. If a positive value is used, the queue will reject entries
+     * once it is full.
+     */
+    static final Setting<Integer> TASK_QUEUE_CAPACITY_SETTING = Setting.intSetting(
+        "xpack.inference.http.request_executor.queue_capacity",
+        2000,
+        0,
```

Comment:
I think we probably should be setting a maximum value here. If the queue size setting is too large, I think it could cause the node to crash.

Reply:
Yeah true, my only concern at this point is the limit will change depending on how much memory the elasticsearch node has and the size of the text stored in the queue. My thinking was that we could just document it with a warning that setting it too high could result in out of memory errors.